### ISSUE TITLE
@angular-eslint/template/eqeqeq: allowNullOrUndefined: true option has no effect in v22

### ISSUE THREAD

**Issue by schallm** [angular-eslint/angular-eslint#3070]
#### Description

The `allowNullOrUndefined: true` option for `@angular-eslint/template/eqeqeq` is silently ignored in `angular-eslint@22.0.0`. The rule continues to report errors on `!= null` and `== null` comparisons even when the option is explicitly configured.

The option is important because `!= null` is a well-established JavaScript/TypeScript idiom that checks for **both** `null` and `undefined` in a single expression. Replacing it with `!== null` is semantically incorrect as it no longer catches `undefined`.

#### Reproduction

**`eslint.config.js`:**
```js
const angular = require('angular-eslint');
const tseslint = require('typescript-eslint');

module.exports = tseslint.config(
  {
    files: ['**/*.html'],
    extends: [
      ...angular.configs.templateRecommended,
    ],
    rules: {
      '@angular-eslint/template/eqeqeq': ['error', { allowNullOrUndefined: true }],
    },
  }
);
```

**`example.component.html`:**
```html
@let value = someSignal();
@if (value != null) {
  <span>{{ value }}</span>
}
```

**Running `npx eslint example.component.html` produces:**
```
1:6  error  Expected `!==` but received `!=`  @angular-eslint/template/eqeqeq
```

**Running `npx eslint --print-config example.component.html`** correctly shows the option is applied:
```json
"@angular-eslint/template/eqeqeq": [
  2,
  { "allowNullOrUndefined": true }
]
```

The resolved config is correct, yet the rule still fires — confirming the bug is in the rule's runtime behaviour, not in config merging.

#### Expected Behaviour

No error reported for `!= null` or `== null` comparisons when `allowNullOrUndefined: true` is set, consistent with the documented intent of the option and the behaviour of ESLint core's `eqeqeq` rule with `"smart"` or `"allow-null"`.

#### Actual Behaviour

The rule fires on `!= null` regardless of the `allowNullOrUndefined` option value.

#### Environment

| Package | Version |
|---|---|
| `angular-eslint` | 22.0.0 |
| `@angular-eslint/eslint-plugin-template` | 22.0.0 |
| `@angular-eslint/bundled-angular-compiler` | 18.4.3 |
| `@angular/compiler` | 22.0.0 |
| `eslint` | 10.4.1 |
| `typescript-eslint` | 8.60.1 |
| Node.js | 24.15.0 |

#### Workaround

Until this is fixed, suppress the rule inline with a comment explaining the reason:

```html
<!-- eslint-disable-next-line @angular-eslint/template/eqeqeq -- allowNullOrUndefined broken in angular-eslint v22 -->
@if (value != null) {
```

**Comment by EB-Teampeak** [comment]
For anyone encountering this issue at a later date, we also encountered this. However for us the `@angular-eslint/bundled-angular-compiler` was being pinned at `18.2.0` which caused the problem. 

For us the `ngxtension-plugin` pinned `18.2.0` which caused the problem. We 'fixed' this by adding an override that forces the package versions to match the `angular-eslint` by adding the following entry to the package.json:
```json
"overrides": {
    "@angular-eslint/bundled-angular-compiler": "$angular-eslint"
  }
```
